libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/datahandling/PersistedStats.java

```diff
@@ -111,14 +131,32 @@ private void sendToListeners(Properties newProps) {
     }
 
     private synchronized Properties writeSyncStats(long time, long bytesSent, long totalObs,
-                                            long totalCells, long totalWifis, long obsPerDay) throws IOException {
-        final FileOutputStream out = new FileOutputStream(mStatsFile);
+                                            long totalCells, long totalWifis, long obsPerDay) {
+        final FileOutputStream out;
         final Properties props = createStatsProp(time, bytesSent, totalObs, totalCells, totalWifis, obsPerDay);
 
+
+        try {
+            out = new FileOutputStream(mStatsFile);
```

Comment:
This could throw prior to assignment to out. Hopefully out was defaulted to null.